app/medInria/medFilteringWorkspace.cpp

```diff
@@ -13,45 +13,26 @@
 
 #include <medFilteringWorkspace.h>
 
-
-#include <medDatabaseNonPersistentController.h>
+#include <medAbstractLayeredView.h>
+#include <medAbstractSelectableToolBox.h>
+#include <medDataManager.h>
 #include <medMetaDataKeys.h>
-#include <medStorage.h>
-#include <medFilteringSelectorToolBox.h>
+#include <medSelectorToolBox.h>
+#include <medTabbedViewContainers.h>
 #include <medToolBoxFactory.h>
 #include <medViewContainer.h>
-#include <medTabbedViewContainers.h>
-#include <medFilteringAbstractToolBox.h>
-#include <medAbstractData.h>
-#include <medDataManager.h>
-#include <medAbstractView.h>
-#include <medAbstractLayeredView.h>
-#include <medViewParameterGroup.h>
-#include <medLayerParameterGroup.h>
 
 class medFilteringWorkspacePrivate
 {
 public:
-    QPointer<medFilteringSelectorToolBox> filteringToolBox;
     medViewContainer *inputContainer;
     medViewContainer *outputContainer;
 
-
     medAbstractData *filterOutput;
 };
 
-medFilteringWorkspace::medFilteringWorkspace(QWidget *parent): medAbstractWorkspace (parent), d(new medFilteringWorkspacePrivate)
+medFilteringWorkspace::medFilteringWorkspace(QWidget *parent): medSelectorWorkspace (parent, name()), d(new medFilteringWorkspacePrivate)
```

Comment:
I know I asked to do it like this but I made a slight error: it should call staticName(), not name() (the latter is a virtual function and should not be called here).
BUT
I think we could do things slightly differently for the workspace initialization because of the registration workspace problem (since it uses a toolbox derived from selectorToolbox). We can talk about this at the meeting.